generic_tracers/generic_COBALT.F90

```diff
@@ -7575,232 +6046,8 @@ subroutine generic_COBALT_update_from_bottom(tracer_list, dt, tau, model_time)
-    phyto(LARGE)%fp_btm = phyto(LARGE)%fp_btm/dt
-    call g_tracer_get_pointer(tracer_list,'plg_btf','field',temp_field)
-    temp_field(:,:,1) = phyto(LARGE)%fp_btm(:,:)
-    call g_tracer_set_values(tracer_list,'plg','btm_reservoir',0.0)
-    if (phyto(LARGE)%id_fp_btm .gt. 0)           &
-         used = g_send_data(phyto(LARGE)%id_fp_btm,phyto(LARGE)%fp_btm, &
-         model_time, rmask = grid_tmask(:,:,1),&
-         is_in=isc, js_in=jsc,ie_in=iec, je_in=jec)
-
-    call g_tracer_get_values(tracer_list,'pmd','btm_reservoir',phyto(MEDIUM)%fp_btm,isd,jsd)
-    phyto(MEDIUM)%fp_btm = phyto(MEDIUM)%fp_btm/dt
-    call g_tracer_get_pointer(tracer_list,'pmd_btf','field',temp_field)
-    temp_field(:,:,1) = phyto(MEDIUM)%fp_btm(:,:)
-    call g_tracer_set_values(tracer_list,'pmd','btm_reservoir',0.0)
-    if (phyto(MEDIUM)%id_fp_btm .gt. 0)           &
-         used = g_send_data(phyto(MEDIUM)%id_fp_btm,phyto(MEDIUM)%fp_btm, &
-         model_time, rmask = grid_tmask(:,:,1),&
-         is_in=isc, js_in=jsc,ie_in=iec, je_in=jec)
-
-    call g_tracer_get_values(tracer_list,'psm','btm_reservoir',phyto(SMALL)%fp_btm,isd,jsd)
-    phyto(SMALL)%fp_btm = phyto(SMALL)%fp_btm/dt
-    call g_tracer_get_pointer(tracer_list,'psm_btf','field',temp_field)
-    temp_field(:,:,1) = phyto(SMALL)%fp_btm(:,:)
-    call g_tracer_set_values(tracer_list,'psm','btm_reservoir',0.0)
-    if (phyto(SMALL)%id_fp_btm .gt. 0)           &
-         used = g_send_data(phyto(SMALL)%id_fp_btm,phyto(SMALL)%fp_btm, &
-         model_time, rmask = grid_tmask(:,:,1),&
-         is_in=isc, js_in=jsc,ie_in=iec, je_in=jec)
-    !
-    ! Handle phytoplankton sinking: silicate
-    !
-
-    call g_tracer_get_values(tracer_list,'silg','btm_reservoir',phyto(LARGE)%fsi_btm,isd,jsd)
-    phyto(LARGE)%fsi_btm = phyto(LARGE)%fsi_btm/dt
-    call g_tracer_get_pointer(tracer_list,'silg_btf','field',temp_field)
-    temp_field(:,:,1) = phyto(LARGE)%fsi_btm(:,:)
-    call g_tracer_set_values(tracer_list,'silg','btm_reservoir',0.0)
-    if (phyto(LARGE)%id_fsi_btm .gt. 0)           &
-         used = g_send_data(phyto(LARGE)%id_fsi_btm,phyto(LARGE)%fsi_btm, &
-         model_time, rmask = grid_tmask(:,:,1),&
-         is_in=isc, js_in=jsc,ie_in=iec, je_in=jec)
-
-    call g_tracer_get_values(tracer_list,'simd','btm_reservoir',phyto(MEDIUM)%fsi_btm,isd,jsd)
-    phyto(MEDIUM)%fsi_btm = phyto(MEDIUM)%fsi_btm/dt
-    call g_tracer_get_pointer(tracer_list,'simd_btf','field',temp_field)
-    temp_field(:,:,1) = phyto(MEDIUM)%fsi_btm(:,:)
-    call g_tracer_set_values(tracer_list,'simd','btm_reservoir',0.0)
-    if (phyto(MEDIUM)%id_fsi_btm .gt. 0)           &
-         used = g_send_data(phyto(MEDIUM)%id_fsi_btm,phyto(MEDIUM)%fsi_btm, &
-         model_time, rmask = grid_tmask(:,:,1),&
-         is_in=isc, js_in=jsc,ie_in=iec, je_in=jec)
-
+    call cobalt_send_diagnostics_bottom(tracer_list,model_time,grid_tmask, &
+                            temp_field,dt,isc,iec,jsc,jec,isd,jsd,phyto,cobalt)
```

Comment:
See comment above concerning the role of this routine in updating the bottom reservoirs and setting the bottom fluxes.  Maybe keep as "update_from_bottom" for now?